include/git2/repository.h

```diff
@@ -746,6 +747,15 @@ GIT_EXTERN(const char *) git_repository_get_namespace(git_repository *repo);
 GIT_EXTERN(int) git_repository_is_shallow(git_repository *repo);
 
 /**
+ * Determine the shallow roots of the repository
+ *
+ * @param out An array of shallow oids. Can be NULL.
```

Comment:
Why can it be `NULL`? Why would you call this function if you did not want the answer?